src/pvrclient-nextpvr.cpp

```diff
@@ -636,7 +636,7 @@ PVR_ERROR cPVRClientNextPVR::GetEpg(ADDON_HANDLE handle, int iChannelUid, time_t
       XBMC->Log(LOG_DEBUG, "Skipping expired EPG data %d %ld %lld",iChannelUid,iStart, iEnd);
       return PVR_ERROR_INVALID_PARAMETERS;
   }
-  sprintf(request, "/service?method=channel.listings&channel_id=%d&start=%d&end=%d", iChannelUid, (int)iStart, (int)iEnd);
+  sprintf(request, "/service?method=channel.listings&channel_id=%d&start=%d&end=%d&genre=all", iChannelUid, (int)iStart, (int)iEnd);
```

Comment:
c++ style casts (consider use of StringUtils::Format in the future)

Reply:
Changed the cast,  I plan a further review to change the DoRequest function to and have the first parameter take a string.